src/main/java/net/mcreator/plugin/PluginLoader.java

```diff
@@ -44,15 +49,12 @@
 	private static final Logger LOG = LogManager.getLogger("Plugin Loader");
 
 	public static PluginLoader INSTANCE;
-
-	public static void initInstance() {
-		INSTANCE = new PluginLoader();
-	}
-
 	private final List<Plugin> plugins;
-
 	private final Reflections reflections;
 
+	private List<Plugin> pluginsToUpdate = new ArrayList<>();
+	private List<String> versions = new ArrayList<>();

```

Comment:
Don't make two lists, but make new object eg. PluginUpdateInfo and store list of these objects instead, this is not Javanic right now

Reply:
Or alternatively, a Map as there are only two values